src/SeleniumLibrary/keywords/window.py

```diff
@@ -0,0 +1,221 @@
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from selenium.common.exceptions import NoSuchWindowException
+
+from SeleniumLibrary.base import keyword, LibraryComponent
+from SeleniumLibrary.locators import WindowManager
+
+
+class WindowKeywords(LibraryComponent):
+
+    def __init__(self, ctx):
+        LibraryComponent.__init__(self, ctx)
+        self._window_manager = WindowManager(ctx)
+
+    @keyword
+    def select_window(self, locator='MAIN'):
+        """Selects browser window matching ``locator``.
+
+        If the window is found, all subsequent commands use the selected
+        window, until this keyword is used again. If the window is not
+        found, this keyword fails. The previous window handle is returned,
+        and can be used to return back to it later.
+
+        Notice that in this context _window_ means a pop-up window opened
+        when doing something on an existing window. It is not possible to
+        select windows opened with `Open Browser`, `Switch Browser` must
+        be used instead. Notice also that alerts should be handled with
+        `Handle Alert` or other alert related keywords.
+
+        The ``locator`` can be specified using different strategies somewhat
+        similarly as when `locating elements` on pages.
+
+        - By default the ``locator`` is matched against window handle, name,
+          title, and URL. Matching is done in that order and the the first
+          matching window is selected.
+
+        - The ``locator`` can specify an explicit strategy by using format
+          ``strategy:value`` (recommended) or ``strategy=value``. Supported
+          strategies are ``name``, ``title`` and ``url``, which match windows
+          using name, title, and URL, respectively. Additionally, ``default``
+          can be used to explicitly use the default strategy explained above.
+
+        - If the ``locator`` is ``NEW`` (case-insensitive), the latest
+          opened window is selected. It is an error if this is the same
+          as the current window.
+
+        - If the ``locator`` is ``MAIN`` (default, case-insensitive),
+          the main window is selected.
+
+        - If the ``locator`` is ``CURRENT`` (case-insensitive), nothing is
+          done. This effectively just returns the current window handle.
+
+        - If the ``locator`` is not a string, it is expected to be a list
+          of window handles _to exclude_. Such a list of excluded windows
+          can be get from `Get Window Handles` prior to doing an action that
+          opens a new window.
+
+        Example:
+        | `Click Link`      | popup1      |      | # Open new window |
```

Comment:
Nitpick, should the tables be same size?

Reply:
Can you clarify?